File: capgemini/batch-3/12-02-24/RotateArray.py


def rotateArray(arr,i):
    stack=[]
    queue = []
    n = len(arr)-i
    for j in range(len(arr)):
        if j < n:
            queue.append(arr[j])
        else: stack.append(arr[j])
    for j in range(len(arr)):
        if len(stack) > 0:
            arr[j] = stack.pop()
        else: arr[j] = queue.pop(0)
    return arr

def main():
    arr = [3,2,4,5,12,45,67,89]
    i = 4
    ans = rotateArray(arr,i)
    print(ans)




main()

File: madurai/S22/27-06-2024/RotateArray.java
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Stack;

public class RotateArray {
    public static void main(String[] args) {
        int[] arr = {3,2,4,-5,6,7,9,12};
        int pi = 3;
        arr = rotateArray(arr,pi);
        System.out.println(Arrays.toString(arr));
    }

    private static int[] rotateArray(int[] arr,int pi) {
        Stack<Integer> st = new Stack<>();
        Queue<Integer> queue = new LinkedList<>();
        for(int i=0;i<arr.length;i++){
            if(i < pi) queue.add(arr[i]);
            else st.push(arr[i]);
        }
        for(int i=0;i<arr.length;i++){
            if(st.size() > 0) arr[i] = st.pop();
            else arr[i] = queue.poll();
        }

        return arr;
    }
}
